spark/spark-scala-parent/src/main/scala/org/apache/zeppelin/spark/BaseSparkScalaInterpreter.scala

```diff
@@ -237,6 +237,13 @@ abstract class BaseSparkScalaInterpreter(val conf: SparkConf,
       case None =>
     }
 
+    val webUiUrl = properties.getProperty("zeppelin.spark.uiWebUrl");
+    if (!StringUtils.isBlank(webUiUrl)) {
+      this.sparkUrl = webUiUrl.replace("{{applicationId}}", sc.applicationId);
+    } else {
+      useYarnProxyURLIfNeeded()
+    }
+
```

Comment:
I would move this into a separate function to remove duplicates.
Another issue - we need to make safe value from `sc.applicationId`, so we couldn't get any security problem...

Reply:
Thanks for the suggestion @alexott , I have removed the code duplicated. What do you mean security problem ? spark appId is generated by spark resource manager ? I don't see security problem

Reply:
Ok, that was my concern - if we can say that `applicationId` is safely generated, and not provided by user, then I'm ok...